datasource.go

```diff
@@ -98,6 +98,10 @@ func (ds *SQLDatasource) QueryData(ctx context.Context, req *backend.QueryDataRe
 
 	wg.Add(len(req.Queries))
 
+	if queryDataMutator, ok := ds.driver().(QueryDataMutator); ok {
+		ctx, req = queryDataMutator.MutateQueryData(ctx, req)
```

Comment:
any reason for returning context?

Reply:
We need to return it if we're doing things to modify it in the mutator function:

https://github.com/grafana/grafana/blob/0a870e6a8855cee9c8ae391767da40dbb1c16cc6/pkg/tsdb/azuremonitor/azuremonitor.go#L370

Example of ADX

Reply:
TIL it creates new context 🤔